src/helpers.tsx

```diff
@@ -0,0 +1,160 @@
+  if (editorElement === null) {
+    return;
+  }
+
+  const componentName: string = getComponentNodeName(editorElement);
+
+  let editorContainer = document.querySelector('#hot-wrapper-editor-container-' + componentName);
+  if (!document.querySelector('#hot-wrapper-editor-container-' + componentName)) {
+    editorContainer = document.createElement('DIV');
+    editorContainer.id = 'hot-wrapper-editor-container-' + componentName;
+    document.body.appendChild(editorContainer);
+  }
+
+  return ReactDOM.createPortal(editorElement, editorContainer);
+}
+
+/**
+ * Get an editor element extended with a instance-emitting method.
+ *
+ * @param {React.ReactNode} children Component children.
+ * @param {Map} editorCache Component's editor cache.
+ * @returns {React.ReactElement} An editor element containing the additional methods.
+ */
+export function getExtendedEditorElement(children: React.ReactNode, editorCache: Map<string, object>): React.ReactElement | null {
+  const editorElement = getChildElementByType(children, 'hot-editor');
+
+  if (!editorElement) {
+    return null;
+  }
+
+  return React.cloneElement(editorElement, {
+    emitEditorInstance: (editorName, editorInstance) => {
+      if (!editorCache.has(editorName)) {
+        editorCache.set(editorName, editorInstance);
+      }
+    },
+    isEditor: true
+  });
+}
+
+/**
+ * Create a react component and render it to an external DOM done.
+ *
+ * @param {React.ReactElement} rElement React element to be used as a base for the component.
+ * @param {Object} props Props to be passed to the cloned element.
+ * @param {Function} callback Callback to be called after the component has been mounted.
+ * @param {Document} [ownerDocument] The owner document to set the portal up into.
+ * @returns {{portal: React.ReactPortal, portalContainer: HTMLElement}} An object containing the portal and its container.
+ */
+export function createPortal(rElement: React.ReactElement, props, callback: Function, ownerDocument: Document = document): {
+  portal: React.ReactPortal,
+  portalContainer: HTMLElement
+} {
+  if (!ownerDocument) {
+    ownerDocument = document;
+  }
+
+  if (!bulkComponentContainer) {
+    bulkComponentContainer = ownerDocument.createDocumentFragment();
+  }
+
+  const portalContainer = ownerDocument.createElement('DIV');
+  bulkComponentContainer.appendChild(portalContainer);
+
+  const extendedRendererElement = React.cloneElement(rElement, {
+    key: `${props.row}-${props.col}`,
+    ...props
+  });
+
+  return {
+    portal: ReactDOM.createPortal(extendedRendererElement, portalContainer, `${props.row}-${props.col}-${Math.random()}`),
+    portalContainer
+  };
+}
+
+/**
+ * Add the `UNSAFE_` prefixes to the deprecated lifecycle methods for React >= 16.3.
+ *
+ * @param {Function} Klass Class to have the methods renamed.
+ * @returns {Function} Class with the renamed methods.
+ */
+export function addUnsafePrefixes(Klass) {
```

Comment:
This makes the typescript definitions largely useless as the generated `hotTable.d.ts` now looks like this:

```ts
declare const PrefixedHotTable: any;
export default PrefixedHotTable;
export { PrefixedHotTable as HotTable };
```